src/renderer/services/wallet/service.test.ts

```diff
@@ -0,0 +1,93 @@
+import { none, some } from 'fp-ts/lib/Option'
+import * as fs from 'fs-extra'
+
+import { removeKeystore, keystoreService, KEY_FILE } from './service'
+
+jest.mock('electron', () => ({ ipcRenderer: { send: jest.fn() } }))
+
+const mockEncrypt = { encrypt: 'info' }
+const mockDecrypt = { phrase: 'phrases' }
+
+jest.mock('@thorchain/asgardex-crypto', () => ({
+  encryptToKeyStore: () => mockEncrypt,
+  decryptFromKeystore: () => Promise.resolve(mockDecrypt.phrase)
+}))
+
+describe('services/keystore/service/', () => {
+  const _setKeystoreStateSpy = jest.spyOn(keystoreService, '_setKeystoreState')
+  const fsRemoveSpy = jest.spyOn(fs, 'remove')
+  const fsWriteJsonSpy = jest.spyOn(fs, 'writeJSON')
+  const fsEnsureFileSpy = jest.spyOn(fs, 'ensureFile')
+  const fsPathExistsSpy = jest.spyOn(fs, 'pathExists')
+  const fsReadJSONSpy = jest.spyOn(fs, 'readJSON')
+
+  beforeEach(() => {
+    fsRemoveSpy.mockImplementationOnce(jest.fn)
+    fsWriteJsonSpy.mockImplementationOnce(jest.fn)
+    fsEnsureFileSpy.mockImplementationOnce(jest.fn)
+  })
+
+  it('removeKeystore', async (done) => {
+    await removeKeystore()
+    expect(fsRemoveSpy).toBeCalledWith(KEY_FILE)
+    expect(_setKeystoreStateSpy).toBeCalledWith(none)
+    done()
+  })
+
+  it('addKeystore', async (done) => {
+    const removeKeystoreSpy = jest.spyOn(keystoreService, 'removeKeystore')
+    const phrase = 'phrase'
+    const password = 'password'
+    removeKeystoreSpy.mockImplementationOnce(() => Promise.resolve())
+
+    await keystoreService.addKeystore(phrase, password)
+
+    expect(removeKeystoreSpy).toBeCalled()
+    expect(fsEnsureFileSpy).toBeCalledWith(KEY_FILE)
+    expect(fsWriteJsonSpy).toBeCalledWith(KEY_FILE, mockEncrypt)
+    expect(_setKeystoreStateSpy).toBeCalledWith(some(some({ phrase })))
+    done()
+  })
+
+  describe('addPhrase', () => {
+    const password = 'password'
+    it('should reject if keystore is not imported', (done) => {
+      let failed = false
+      keystoreService
+        .unlock(none, password)
+        .catch(() => {
+          failed = true
+        })
+        .then(() => {
+          expect(failed).toBeTruthy()
+          done()
+        })
+    })
+
+    it('should reject if keystore file does not exist', (done) => {
+      let failed = false
+
+      fsPathExistsSpy.mockImplementationOnce(() => Promise.resolve(false))
+
+      keystoreService
+        .unlock(some(none), password)
+        .catch(() => {
+          failed = true
+        })
+        .then(() => {
+          expect(failed).toBeTruthy()
+          done()
+        })
+    })
+
+    it('should call setKeystoreState', async (done) => {
+      fsPathExistsSpy.mockImplementationOnce(() => Promise.resolve(true))
+      fsReadJSONSpy.mockImplementationOnce(() => Promise.resolve(mockDecrypt))
+
+      await keystoreService.unlock(some(some({ phrase: 'phrase' })), password)
+
+      expect(keystoreService._setKeystoreState).toBeCalledWith(some(some(mockDecrypt)))
```

Comment:
It looks like a similar test as you already have done in  `addKeystore` (line 47), isn't?

Reply:
not exactly. this tests that `addPhrase` calls `setKeystoreState` with appropriate wrguments